azurerm/data_source_redis_cache.go

```diff
@@ -0,0 +1,266 @@
+							Optional:  true,
+							Sensitive: true,
+						},
+						"enable_authentication": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+					},
+				},
+			},
+
+			"patch_schedule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"day_of_week": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"start_hour_utc": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"hostname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"ssl_port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"primary_access_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"secondary_access_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"tags": tagsForDataSourceSchema(),
+		},
+	}
+}
+
+func dataSourceArmRedisCacheRead(d *schema.ResourceData, meta interface{}) error {
+	ctx := meta.(*ArmClient).StopContext
+	client := meta.(*ArmClient).redisClient
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: Redis instance %q (Resource group %q) was not found", name, resourceGroup)
+		}
+		return fmt.Errorf("Error reading the state of Redis instance %q: %+v", name, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if zones := resp.Zones; zones != nil {
+		d.Set("zones", zones)
+	}
+
+	if sku := resp.Sku; sku != nil {
+		d.Set("capacity", sku.Capacity)
+		d.Set("family", sku.Family)
+		d.Set("sku_name", sku.Name)
+	}
+
+	if props := resp.Properties; props != nil {
+		d.Set("ssl_port", props.SslPort)
+		d.Set("hostname", props.HostName)
+		d.Set("minimum_tls_version", string(props.MinimumTLSVersion))
+		d.Set("port", props.Port)
+		d.Set("enable_non_ssl_port", props.EnableNonSslPort)
+		if props.ShardCount != nil {
+			d.Set("shard_count", props.ShardCount)
+		}
+		d.Set("private_static_ip_address", props.StaticIP)
+		d.Set("subnet_id", props.SubnetID)
+	}
+
+	redisConfiguration, err := flattenRedisConfiguration(resp.RedisConfiguration)
+	if err != nil {
+		return fmt.Errorf("Error flattening `redis_configuration`: %+v", err)
+	}
+	if err := d.Set("redis_configuration", redisConfiguration); err != nil {
+		return fmt.Errorf("Error setting `redis_configuration`: %+v", err)
+	}
+
+	keys, err := client.ListKeys(ctx, resourceGroup, name)
+	if err != nil {
+		return err
+	}
+
+	d.Set("primary_access_key", *keys.PrimaryKey)
+	d.Set("secondary_access_key", *keys.SecondaryKey)
```

Comment:
can we remove this pointer, `d.Set` will automatically do this (and handle it being nil)

```suggestion
	d.Set("secondary_access_key", keys.SecondaryKey)
```